code/datums/mapgen/biomes/_biome.dm

```diff
@@ -0,0 +1,50 @@
+///This datum handles the transitioning from a turf to a specific biome, and handles spawning decorative structures and mobs.
+/datum/biome
+	///Type of turf this biome creates
+	var/turf_type
+	///Chance of having a structure from the flora types list spawn
+	var/flora_density = 0
+	///Chance of having a mob from the fauna types list spawn
+	var/fauna_density = 0
+	///list of type paths of objects that can be spawned when the turf spawns flora
+	var/list/flora_types = list(/obj/tree1)
+	///list of type paths of mobs that can be spawned when the turf spawns fauna
+	var/list/fauna_types = list()
```

Comment:
would love for these to be weighted assoc. lists

Reply:
added